### ISSUE TITLE
Improperly throws System.TypeLoadException on mock when a record has a base record on dotnet 6

### ISSUE THREAD

**Issue by tgrieger-sf** [devlooped/moq#1273]
A System.TypeLoadException is incorrectly thrown on dotnet 6 when mocking a record with a base record. See simple repro below.

```C#
using System;
using Moq;

namespace Testing
{
	public record First : Base
	{
		public virtual string Something { get; init; }
	}

	public record Base {}

	class Program
	{
		static void Main(string[] args)
		{
			Mock<First> mockFirst = new();
			mockFirst.SetupGet(f => f.Something).Returns("string");
			Console.WriteLine(mockFirst.Object.Something);
		}
	}
}

```

**Comment by stakx** [comment]
The Entity Framework team noticed the same thing and reported it directly to the DynamicProxy project (upon which Moq 4 is built): https://github.com/castleproject/Core/issues/601.

This has recently been fixed there, and it will be fixed in Moq once we can upgrade to the upcoming version of DynamicProxy.

**Comment by tgrieger-sf** [comment]
Awesome, thanks for the update!

**Comment by tgrieger-sf** [comment]
Thanks @stakx! 🚀

**Comment by stakx** [comment]
You're welcome @tgrieger-sf. FYI, I'm going to push a new release to NuGet shortly.

**Comment by stakx** [comment]
This is now fixed in Moq version 4.18.2, which has just been pushed to NuGet.
